usr/share/rear/lib/_input-output-functions.sh

```diff
@@ -1069,6 +1069,13 @@ function cleanup_build_area_and_end_program () {
 #   which lets a 'UserInput -I BAR_CHOICE' call autorespond with 'third choice'.
 #   No USER_INPUT_BAR_CHOICE variable should exist to get real user input for a 'UserInput -I BAR_CHOICE' call
 #   or the user can interupt any automated response within a relatively short time (minimum is only 1 second).
+# * The non-interactive feature works by erroring out in UserInput if the same user input ID is used more than once and
+#   if the first call to UserInput was not met by a human response. The idea is that if the first call ran into the
+#   timeout then the second call will not have a better chance of success and should be treated as an error.
+#   That way, ReaR will quickly abort in situations where human intervention is required but not provided.
+#   This is implemented via an associative Bash array USER_INPUT_SEEN_WITH_TIMEOUT which tracks which user input IDs
```

Comment:
If you are going to use an associative array, it is time to discontinue support for Bash 3 (#2941)

Reply:
Didn't we want to do that for ReaR **2.8** or following in any case?

With `tools/run-in-docker` I checked that none of the distros that we want to support actively uses Bash 3.x, CentOS 6 is the oldest with Bash 4.1 (and I really would like to put out the question if we need to support CentOS 6 with **new** releases)

Reply:
Sure, I don't object to it

Reply:
I wonder why an associative array is required here#
because as far as I see only
```
USER_INPUT_SEEN_WITH_TIMEOUT[$user_input_ID]=true
```
is used so all array element values are same 'true'.
I think when all array element values are same
their values have no information so instead of
an associative array whith useless element values
one could use a nomal indexed array where its element
values are the key values of the associative array like
```
is_true "$NON_INTERACTIVE" && USER_INPUT_SEEN_WITH_TIMEOUT+=( "$user_input_ID" )
```
and
```
if is_true "$NON_INTERACTIVE" ; then
    if IsInArray "$user_input_ID" "${USER_INPUT_SEEN_WITH_TIMEOUT[@]}" 2>/dev/null; then
        Error "UserInput: non-interactive mode and repeat input for '$user_input_ID' requested while the previous attempt was answered with the default or timed out"
```
As far as I see duplicate $user_input_ID element values do not matter
because it only matters whether or not a user_input_ID appears
at least once.

Reply:
@jsmeix you are right, very good thinking. I'll rework the code to use a regular array instead.